Remotion/Data/DomainObjects.Validation.IntegrationTests/Testdomain/StubStorageProvider.cs

```diff
@@ -23,80 +23,84 @@
 using Remotion.Data.DomainObjects.Persistence.Configuration;
 using Remotion.Data.DomainObjects.Persistence.Rdbms.DataReaders;
 using Remotion.Data.DomainObjects.Queries;
-using Remotion.Data.DomainObjects.Tracing;
 using Remotion.Utilities;
 
 namespace Remotion.Data.DomainObjects.Validation.IntegrationTests.Testdomain
 {
-  public class StubStorageProvider : StorageProvider
+  public class StubStorageProvider : IStorageProvider
   {
     public const string StorageProviderID = "StubStorageProvider";
-    public const string GetSecurableObjectsQueryID = "StubQueryID";
+    private bool _disposed;
+    private StorageProviderDefinition _storageProviderDefinition;
 
-    public StubStorageProvider (StorageProviderDefinition definition, IPersistenceExtension persistenceExtension)
-      : base(definition, persistenceExtension)
+    public StubStorageProvider (StorageProviderDefinition definition)
     {
+      ArgumentUtility.CheckNotNull("definition", definition);
+
+      _storageProviderDefinition = definition;
     }
 
-    public override ObjectLookupResult<DataContainer> LoadDataContainer (ObjectID id)
+    public StorageProviderDefinition StorageProviderDefinition => _storageProviderDefinition;
```

Comment:
as you like: you could make this a readonly auto-property  (moved to before the ctor next to the other state.